pkg/apply/applier_apply_options.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package apply
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cli-utils/pkg/apply/event"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+)
+
+const defaultPollInterval = 2 * time.Second
+
+type ApplierOptions struct {
+	// Encapsulates the fields for server-side apply.
+	serverSideOptions common.ServerSideOptions
+
+	// reconcileTimeout defines whether the applier should wait
+	// until all applied resources have been reconciled, and if so,
+	// how long to wait.
+	reconcileTimeout time.Duration
+
+	// pollInterval defines how often we should poll for the status
+	// of resources.
+	pollInterval time.Duration
+
+	// emitStatusEvents defines whether status events should be
+	// emitted on the eventChannel to the caller.
+	emitStatusEvents bool
+
+	// prune defines whether pruning of previously applied
+	// objects should happen after apply.
+	prune bool
+
+	// dryRunStrategy defines whether changes should actually be performed,
+	// or if it is just talk and no action.
+	dryRunStrategy common.DryRunStrategy
+
+	// prunePropagationPolicy defines the deletion propagation policy
+	// that should be used for pruning. If this is not provided, the
+	// default is to use the Background policy.
+	prunePropagationPolicy metav1.DeletionPropagation
+
+	// pruneTimeout defines whether we should wait for all resources
+	// to be fully deleted after pruning, and if so, how long we should
+	// wait.
+	pruneTimeout time.Duration
+
+	// inventoryPolicy defines the inventory policy of apply.
+	inventoryPolicy inventory.InventoryPolicy
+
+	// eventListeners is a list of event listeners that are called in specified order when an event happens.
+	// Listeners are never called concurrently with each other.
+	eventListeners []func(event.Event)
+}
+
+func NewApplierOptions() *ApplierOptions {
```

Comment:
Why would we have a pointer to the ApplierOptions? With the ApplierOptions just holding data, it seems unnecessary.